node/src/header_ex.rs

```diff
@@ -320,7 +383,14 @@ impl Codec for HeaderCodec {
         }
 
         if msgs.is_empty() {
-            return Err(io::Error::new(io::ErrorKind::Other, "invalid response"));
+            // We have two cases that can reach here:
```

Comment:
```suggestion
            // There are two cases we can reach here:
```
or maybe "We have two cases that we can reach here"